code/modules/xenoarchaeology/xenoartifact_traits.dm

```diff
@@ -0,0 +1,1011 @@
+	var/obj/item/xenoartifact/X = source
+	if(istype(thing, /obj/item/stock_parts/cell))
+		var/obj/item/stock_parts/cell/C = thing
+		if(X.default_activate(charge, user, user))
+			C.charge -= X.charge_req*10
+
+//Minor traits - Use the to define aspects of the artifact without any immediate interaction
+/datum/xenoartifact_trait/minor/looped
+	desc = "Looped"
+	label_desc = "Looped: The Artifact feeds into itself and amplifies its own charge."
+
+/datum/xenoartifact_trait/minor/looped/on_item(obj/item/xenoartifact/X, atom/user, atom/item)
+	if(istype(item, /obj/item/multitool))
+		to_chat(user, "<span class='info'>The [item.name] displays a resistance reading of [X.charge_req*0.1].</span>") 
+		return TRUE
+	..()
+
+/datum/xenoartifact_trait/minor/looped/activate(obj/item/xenoartifact/X)
+	X.charge = ((100-X.charge)*0.2)+X.charge //This should generally cut off around 100
+
+/datum/xenoartifact_trait/minor/capacitive
+	desc = "Capacitive"
+	label_desc = "Capacitive: The Artifact's structure allows it to hold extra charges."
+	var/charges //Extra uses, not total
+	var/saved_cooldown //This may be considered messy but it's a more practical approach that avoids making an edgecase
+
+/datum/xenoartifact_trait/minor/capacitive/on_init(obj/item/xenoartifact/X)
+	charges = pick(0, 1, 2) //Extra charges, not total
+
+/datum/xenoartifact_trait/minor/capacitive/on_touch(obj/item/xenoartifact/X, mob/user)
+	to_chat(user, "<span class='notice'>The hairs on your neck stand up after touching the [X.name].</span>")
+	return TRUE
+
+/datum/xenoartifact_trait/minor/capacitive/activate(obj/item/xenoartifact/X)
+	if(!(saved_cooldown) && X.cooldown)
+		saved_cooldown = X.cooldown //Avoid doing this on init beacause malfunctions can change it in the future
+	if(charges)
+		charges -= 1
+		X.cooldown = -1000 SECONDS //This is better than making a unique interaction in xenoartifact.dm
+		return
+	charges = pick(0, 1, 2)
+	X.cooldown = saved_cooldown
+	saved_cooldown = null
+
+/datum/xenoartifact_trait/minor/capacitive/on_item(obj/item/xenoartifact/X, atom/user, atom/item)
+	if(istype(item, /obj/item/multitool))
+		to_chat(user, "<span class='info'>The [item.name] displays an overcharge reading of [charges/3].</span>") 
+		return TRUE
+	..()
+
+/datum/xenoartifact_trait/minor/dense //Rather large, quite gigantic, particularly big
+	desc = "Dense"
+	label_desc = "Dense: The Artifact is dense and cannot be easily lifted but, the design has a slightly higher reaction rate."
+	blacklist_traits = list(/datum/xenoartifact_trait/minor/wearable, /datum/xenoartifact_trait/minor/sharp, /datum/xenoartifact_trait/minor/light, /datum/xenoartifact_trait/minor/heavy, /datum/xenoartifact_trait/minor/blocking)
+
+/datum/xenoartifact_trait/minor/dense/on_init(obj/item/xenoartifact/X)
+	X.density = TRUE
+	X.interaction_flags_atom = INTERACT_ATOM_ATTACK_HAND
+	X.interaction_flags_item = INTERACT_ATOM_ATTACK_HAND
```

Comment:
What does this do?

Reply:
This stop changes the interaction flags to disallow people picking up the artifact. This is from the dense trait, that makes the artifact mimic a heavy structure.